### PR TITLE
Eds bugz

### PR DESCRIPTION
**Pull request by roblevintennis**
**Description:**
- [Asana Task](https://app.asana.com/0/1150068311310825/1158385663233977/f)
- Fix Safari XHR request.setTimeout undefined error
- Fix sort column header console error
- Indirectly fixes no window.fetch support in IE by giving notice to consumers how to polyfill

The was also tested by pulling in from https://github.com/getethos/ethos/pull/2062 (Nora)

**Screenshots:**

BEFORE (safari):

image

AFTER (safari):

image

### CODE REVIEW

File: src/components/Grid/Position.js [hunk 1]
```diff
@@ -42,8 +42,9 @@ export const Position = (props) => {
     let x
 
     const y = props.refs.findIndex((columnRefs) => {
-      x = columnRefs.findIndex((columnRef) =>
-        columnRef.current.contains(event.target)
+      x = columnRefs.findIndex(
+        (columnRef) =>
+          columnRef.current && columnRef.current.contains(event.target)
```
Comment (roblevintennis) [inline src/components/Grid/Position.js]: This fixes `Fix sort column header console error` discovered in Nora

File: src/xhr/components/content.md [hunk 2]
```diff
@@ -1 +1,20 @@
 The section houses Xhr components for making HTTP requests and handling server errors gracefully.
+
+**_NOTE: Consumers will need to install wathwg-fetch pollyfill to work in IE_**
```
Comment (roblevintennis) [inline src/xhr/components/content.md]: I discovered we polyfill this in [monorepo](https://github.com/getethos/ethos/blob/trunk/frontend/shared/webApp/polyfill.js#L7) already, but not yet in Nora. I believe Tim was well aware of this as he added the polyfill and was original author of XHR, but we hadn't had a chance to discuss. So I think this is an important notice for consumers to see given the nuance that Babel does not give you this for free!

File: src/xhr/lib/xhr.js [hunk 3]
```diff
@@ -19,8 +19,23 @@ async function xhr(options) {
...
+         *
+         * We try to extend window.Request with setTimestamp but it appears to
+         * be disallowed in Safari:
+         * TypeError - request.setTimestamp is not a function
+         *
+         * However, we appear to be able to arbitrarily set and get props in Safari:
+         *   window.Request.timestamp = 1234
+         *   window.Request.timestamp // outputs 1234
+         */
+        if (request.setTimestamp) {
```
Comment (roblevintennis) [inline src/xhr/lib/xhr.js]: Fixes Safari XHR request.setTimeout undefined error

File: src/xhr/src/xhr.ts [hunk 4]
```diff
@@ -31,8 +31,23 @@ async function xhr(options: XhrOptions) {
...
+     * We try to extend window.Request with setTimestamp but it appears to
+     * be disallowed in Safari:
+     * TypeError - request.setTimestamp is not a function
+     *
+     * However, we appear to be able to arbitrarily set and get props in Safari:
+     *   window.Request.timestamp = 1234
+     *   window.Request.timestamp // outputs 1234
+     */
+
+    if (request.setTimestamp) {
```
Comment (roblevintennis) [inline src/xhr/src/xhr.ts]: Fixes Safari XHR request.setTimeout undefined error

File: src/xhr/components/content.md (hunk 2 above)
Comment (zach-ethos) [reply src/xhr/components/content.md]: * whatwg-fetch

**Comment by roblevintennis** [comment]
I consumed this from Nora and tried clicking off the column just off the sort anchor and no more errors:

image

File: src/components/Grid/Position.js (hunk 1 above)
Comment (roblevintennis) [reply src/components/Grid/Position.js]: image

above is original error. Here's after no error after clicking just outside of the sort field (not clicking directly on the sort icons/anchor was the repro):

image

File: src/xhr/components/content.md (hunk 2 above)
Comment (roblevintennis) [reply src/xhr/components/content.md]: thanks, fixed in 5b9376e

File: src/xhr/src/xhr.ts (hunk 4 above)
Comment (roblevintennis) [reply src/xhr/src/xhr.ts]: Ok, I pulled this into nora and ran on safari no more `setTimestamp` bug:

image

Comment (stephaniel96) [reply src/xhr/src/xhr.ts]: woot woot!

**Pull request merged**